docs/mocksupport.rst

```diff
@@ -414,104 +391,247 @@ of the definition and execution of InvokeMethod.
 ----------------------------
 
-Having data with which to respond to requests from a client is required for
-the faked connection to do useful work.  Therefore, in incorporates a
-repository which can be built as part of a test.
+The following methods can be used to add CIM instances, classes and
+qualifier types to the mock repository of a
+:meth:`~pywbem_mock.FakedWBEMConnection` object:
+
+* :meth:`~pywbem_mock.FakedWBEMConnection.add_cimobjects` adds
+  the specified :ref:`CIM objects` to the mock repository.
 
-There are three methods in the faked connection to add data to the
-mock repository
+  The reason for having this method in addition to `CreateInstance` is that it
+  performs less checks.
 
-1. Add pywbem CIM Objects directly to the repository.
-   This is the method :meth:`~pywbem_mock.FakedWBEMConnection.add_cimobjects`. The
-   goal of this operation is to be able to add objects independently of
-   the normal critera of what is required.  Thus you can add instances withou
-   having corresponding classes in the repository or add classes without
-   having the corresponding qualifier declarations.
+* :meth:`~pywbem_mock.FakedWBEMConnection.compile_mof_str` and
+  :meth:`~pywbem_mock.FakedWBEMConnection.compile_mof_file` compile MOF in
+  a string or file, respectively, and add the resulting CIM objects to the
+  mock repository.
 
-2. Compile pywbem objects from strings or file input with the pywbem MOF compiler
-   from strings.
-   These  methods allow the user to build correctly a  defined repository
-   easily simply by defing the MOF for the objects to be inserted in the
-   repository. In order to compile correctly each object type to be compiled
-   must be able to find any prerequisites in the repository. Thus to compile
-   MOF instances, the CIMClasses for those instances must be in the repository
-   and to compile MOF CIM classes the CIM qualifier declarations must be in
-   the repository already.
+  These methods allow the user to easily build a consistent mock repository
+  by defing the MOF for the objects to be inserted into the repository.
 
-   There are two different methods and the only difference is whether they
-   compile from a string input for the MOF or from MOF in a file.
+In all cases, the existing mock repository content is used for resolving any
+prerequisite objects:
 
-   - The method :meth:`~pywbem_mock.FakedWBEMConnection.compile_mof_file` compiles
-     MOF into the repository from a file.
+* For a CIM class to be added, the superclass of that class as well as CIM
+  qualifier types for the qualifiers used by that class must exist.
 
-   - The method :meth:`~pywbem_mock.FakedWBEMConnection.compile_mof_str` compiles
-     MOF into the repository from python string.
+* For a CIM instance to be added, its CIM creation class must exist.
 
+Some CIM objects that are used by a CIM object to be added do not need to
+exist in the mock repository, which in fact allows forward references:
```

Comment:
COMMENT: This is where if we implement issue #1160 something breaks.  Our compiler simply does not check for reference classes rather than allow forward references.

Reply:
OK